run.go

```diff
@@ -326,15 +332,63 @@ func Log(c chan *SimulationStatus) DomainManipulator {
 	}
 }
 
-func (d *InMAP) checkOutputNames(n ...string) error {
+func removeDuplicates(s []string) []string {
+	result := make([]string, 0, len(s))
+	seen := make(map[string]string)
+	for _, val := range s {
+		if _, ok := seen[val]; !ok {
+			result = append(result, val)
+			seen[val] = val
+		}
+	}
+	return result
+}
+
+// checkForDerivitives creates a slice of unique model variables that are
+// required to calculate the requested outputVariables.
```

Comment:
Please add some information to this comment explaining what the inputs and outputs represent.